ecc_logic.py

```diff
@@ -339,6 +339,13 @@ def save_morph_to_appearance(morph_list, appearance):
 
 def dedupe_morphs(morphlists):
     """ removes duplicate morphs from each morphlist in morphlists """
+

```

Comment:
Checked the code, it's not working correctly. I do think it's not far off though. But also, the readability is an issue. It's hard to understand what's going on.